src/lib/auth.ts

```diff
@@ -255,10 +255,27 @@ export const authenticate = (appMode: AppMode, appConfig: AppConfigModelType, ur
       if (!fakeClass || !fakeUser) {
         return reject("Missing fakeClass or fakeUser parameter for demo!");
       }
-      const [userType, userId] = fakeUser.split(":");
+      const userType = fakeUser.split(":")[0];
+      let userId = fakeUser.split(":")[1];
+
       if (((userType !== "student") && (userType !== "teacher")) || !userId) {
         return reject("fakeUser must be in the form of student:<id> or teacher:<id>");
       }
+
+      if ((userId === "random")) {
+        const url = window.location.toString();
+        const title = document.title;
+        const randomStudentId = uniqueId();
+        fakeUser = `student:${randomStudentId}`;
+        userId = randomStudentId;
+        const newUrl = url.replace(/student:random/, fakeUser);
+        if (window.history.state) {
+          window.history.pushState(title, title, newUrl);
+        } else {
+          window.history.replaceState(title, title, newUrl);
+        }
```

Comment:
Why is this conditional on `window.history.state`?  
I'd think we'd always want to use replaceState.  I believe with `pushState` the user could hit the back button and get a URL that had `student:random` in it again, but I don't think it would execute this code again.  So the URL would say `student:random` but really they'd be saving work with a specific user.